src/Psalm/Report/ByIssueLevelAndTypeReport.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Psalm\Report;
+
+use Psalm\Config;
+use Psalm\Internal\Analyzer\DataFlowNodeData;
+use Psalm\Internal\Analyzer\IssueData;
+use Psalm\Report;
+
+use function basename;
+use function get_cfg_var;
+use function ini_get;
+use function strlen;
+use function strtr;
+use function substr;
+use function usort;
+
+final class ByIssueLevelAndTypeReport extends Report
+{
+    /** @var string|null */
+    private $link_format;
+
+    public function create(): string
+    {
+        $this->sortIssuesByLevelAndType();
+
+        $output = <<<HEADING
+|----------------------------------------------------------------------------------------|
+|    Issues have been sorted by level and type. Feature-specific issues and the          |
+|    most serious issues that will always be reported are listed first, with             |
+|    remaining issues in level order. Issues near the top are usually the most serious.  |
+|    Reducing the errorLevel in psalm.xml will suppress output of issues further down    |
+|    this report.                                                                        |
+|                                                                                        |
+|    The level at which issue is reported as an error is given in brackets - e.g.        |
+|    `ERROR (2): MissingReturnType` indicates that MissingReturnType is only reported    |
+|    as an error when Psalm's level is set to 4 or below.                                |
```

Comment:
Is it level 2 or level 4?

Reply:
It's level 2, thanks for catching that. Fixed in 54239838f.